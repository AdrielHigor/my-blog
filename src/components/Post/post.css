button#change-theme {
  padding: 5px 10px;
  border-radius: 4px;
  background: #25282a;
  color: #ffffff;
  font-size: 14px;
  cursor: pointer;
}

div#post {
  margin: auto;
  width: auto;
  max-width: 1080px;
  height: auto;
}

div#post img {
  max-width: 1080px;
}

div#post-content p {
  text-align: start;
  padding-left: 10px;
  font-size: 18px;
}

div#rating {
  border-top: 1px solid #000000;
  margin: 20px;
  margin-bottom: 0px;
  display: flex;
  flex-direction: row;
}

div#like,
#dislike {
  display: flex;
  flex-direction: column;
  text-align: center;
}

div#rating img {
  margin: 10px;
  width: 30px;
}
