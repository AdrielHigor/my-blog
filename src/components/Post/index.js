import React, {
  useState,
  useEffect,
  useReducer,
  useRef,
  useContext
} from "react";
import { useParams } from "react-router-dom";
import firebase from "../../firebase";
import "./post.css";

const initialState = { count: 0 };

function reducer(state, action) {
  switch (action.type) {
    case "like":
      return { count: state.count + 1 };
    case "dislike":
      return { count: state.count - 1 };
    default:
      throw new Error();
  }
}

export default function Post() {
  const { postId } = useParams();
  const [post, setPost] = useState([]);
  const [themeBtnTxt, setTxt] = useState('Darkmode')
  const [state, dispatch] = useReducer(reducer, initialState);
  const likeImg = useRef(null);
  const dislikeImg = useRef(null);
  const backgroundDiv = useRef(null)

  const themes = {
    light: {
      background: "#eeeeee"
    },
    dark: {
      background: "#222222"
    }
  };

  const LightThemeContext = React.createContext(themes.light);
  const DarkThemeContext = React.createContext(themes.dark)

  const LightTheme = useContext(LightThemeContext);
  const DarkTheme = useContext(DarkThemeContext);

  const onLikeClick = () => {
    dislikeImg.current.src =
      "https://img.icons8.com/ios/50/000000/thumbs-down.png";
    likeImg.current.src =
      "https://img.icons8.com/ios-filled/50/000000/thumb-up.png";

    dispatch({ type: "like" });
  };

  const onDislikeClick = () => {
    likeImg.current.src = "https://img.icons8.com/ios/50/000000/thumb-up.png";
    dislikeImg.current.src =
      "https://img.icons8.com/ios-filled/50/000000/thumbs-down.png";

    dispatch({ type: "dislike" });
  };

  useEffect(() => {
    firebase.app.ref("posts").once("value", snapshot => {
      snapshot.forEach(childItem => {
        if (childItem.key === postId) {
          setPost({
            key: childItem.key,
            description: childItem.val().description,
            author: childItem.val().author,
            title: childItem.val().title,
            image: childItem.val().image
          });
        }
      });
    });
  }, []);

  const changeTheme = () => {
    if(themeBtnTxt === 'Darkmode'){
      backgroundDiv.current.style.background = DarkTheme.background
      backgroundDiv.current.style.color = '#ffffff'
      setTxt('Lightmode')
    } else {
      backgroundDiv.current.style.background = LightTheme.background
      backgroundDiv.current.style.color = '#000000'
      setTxt('Darkmode')
    }
  };

  return (
    <div ref={backgroundDiv} style={{ background: LightTheme.background }}>
      <button id="change-theme" onClick={changeTheme}>{themeBtnTxt}</button>
      <div id="post">
        <div id="title">
          <strong id="post-title">{post.title}</strong>
          <span id="post-author">Autor: {post.author}</span>
        </div>
        <img id="post-image" src={post.image} alt="" />
        <div id="post-content">
          <p>{post.description}</p>
        </div>
        <div id="rating">
          <div id="like">
            <img
              ref={likeImg}
              onClick={onLikeClick}
              src="https://img.icons8.com/ios/50/000000/thumb-up.png"
            ></img>
            <span>{state.count}</span>
          </div>
          <div id="dislike">
            <img
              ref={dislikeImg}
              onClick={onDislikeClick}
              src="https://img.icons8.com/ios/50/000000/thumbs-down.png"
            ></img>
          </div>
        </div>
      </div>
    </div>
  );
}
