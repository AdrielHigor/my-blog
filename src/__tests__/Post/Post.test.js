import React from "react";
import Post from "../../components/Post";
import { shallow } from "enzyme";
import "../../setupTests";

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
    useParams: () => ({
        postId: 'post-id1',
    }),
    useRouteMatch: () => ({ url: '/post/post-id1' }),
}));

test("Post rendering correctly", () => {
    const component = shallow(<Post />)

    expect(component.html()).toMatchSnapshot();
}) 