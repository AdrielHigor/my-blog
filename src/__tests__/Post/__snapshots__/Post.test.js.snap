// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post rendering correctly 1`] = `"<div style=\\"background:#eeeeee\\"><button id=\\"change-theme\\">Darkmode</button><div id=\\"post\\"><div id=\\"title\\"><strong id=\\"post-title\\"></strong><span id=\\"post-author\\">Autor: </span></div><img id=\\"post-image\\" alt=\\"\\"/><div id=\\"post-content\\"><p></p></div><div id=\\"rating\\"><div id=\\"like\\"><img src=\\"https://img.icons8.com/ios/50/000000/thumb-up.png\\"/><span>0</span></div><div id=\\"dislike\\"><img src=\\"https://img.icons8.com/ios/50/000000/thumbs-down.png\\"/></div></div></div></div>"`;
